### PR TITLE
feat(reward-service): add videohpsv3 scorer for T2V rollouts

### PR DESCRIPTION
**Pull request by NancyFyong**
## Summary

Adds a `videohpsv3` reward scorer to `unirl-reward-service`, so WAN2.1 T2V
rollouts can be scored with HPSv3 the way Flash-GRPO does: decode the clip,
score every frame against the prompt, and average the best `top_ratio` fraction
of the per-frame scores into one reward.

Flash-GRPO splits this across the wire — its client extracts frames,
JPEG-compresses them, and POSTs each video's frames to a dedicated HPSv3 server
that only ever sees "a batch of images". Here both halves live in one scorer,
matching this repo's convention that a reward name maps to a single
self-contained scorer. `HPSv3Scorer` (added in #142/#143) is reused by
composition, so it keeps owning checkpoint resolution, `max_batch_size` chunking
and the inter-chunk `empty_cache()`; this module adds only decode and aggregation.

Three commits:

1. `feat(reward-service): accept an HF repo id in hpsv3 weights_path`
2. `feat(reward-service): add videohpsv3 scorer for T2V rollouts`
3. `build(reward-service): pin trl, tensorboard and a FlashAttention-2 wheel for hpsv3`

Ordered so the repo-id support lands before the config that depends on it, which
keeps the range bisect-safe.

Points worth a reviewer's attention:

- **`frame_stride` defaults to 1, deliberately.** Raising it changes the
  estimator rather than making it cheaper. A 64-clip step of 81-frame videos
  scores 5184 frames at stride 1 but only 704 at stride 8, and `top_ratio_mean`
  then keeps 3 frames instead of 24 — so a policy can maximise reward by making
  three frames pretty and letting the other 78 rot. We saw exactly that failure
  mode on a long stride-8 run. The cost of stride 1 is affordable: the reward
  call measured **8.2s of a 1029s training step (0.8%)**, so generation dominates.
- **`weights_path` now accepts an HF repo id.** It previously handled only a
  directory holding `HPSv3.safetensors` or the file itself, and passed anything
  else through as a file path — so a repo id reached `torch.load("MizzenAI/HPSv3")`
  and raised, and every deployment had to pre-download 16.5 GB and hardcode a
  local path. Resolution moved into a `_resolve_checkpoint_path` staticmethod
  (mirroring `HPSv2Scorer`) that falls back to `hf_hub_download`, which is the
  same interpretation `hpsv3/inference.py` gives its own default. Hub errors are
  re-raised as `FileNotFoundError` naming both interpretations, because a typo'd
  path otherwise surfaces as an `HFValidationError` about repo id syntax.
- **Per-item failures return NaN rather than raising**, so one undecodable clip
  does not drop a whole 64-item bucket. Missing or mistyped video input still
  raises, because that is a wiring bug and must not be masked.
- **`cv2` is imported lazily**: `registry._try_import` swallows `ImportError` into
  a warning, so a top-level import would silently leave the reward unregistered.
- `extract_frames` and the shared HPSv3 batching helper move into
  `scorers/_common.py`, so `videoalign` (#97) and `video_hpsv3` stop keeping
  separate frame-decode paths. `videoalign` behaviour is unchanged.

## Related Issue

N/A

## Test Plan

Branch is rebased onto `upstream/main` at 5d034c3; the range is `upstream/main..HEAD`.

```
SKIP=no-commit-to-branch pre-commit run --from-ref upstream/main --to-ref HEAD
    # recipe _target_ paths resolve ... Passed (rest skipped: nested project)

ruff check reward_service/scorers/{video_hpsv3,_common,registry,videoalign,hpsv3_scorer}.py
    # All checks passed!

python -c "from reward_service.config import load_config; \
           print(load_config('configs/videohpsv3_service.yaml').rewards[0].params)"
    # {'weights_path': 'MizzenAI/HPSv3', 'config_path': None,
    #  'max_batch_size': 4, 'frame_stride': 1, 'top_ratio': 0.3}
```

`ruff check .` over the whole package reports 33 errors, all pre-existing on
`main` and untouched here; every file this PR adds or modifies is clean.

**Unit tests exist but are not included in this diff.** 50 cases across
`tests/scorers/test_video_hpsv3.py` (frame sampling, top-ratio aggregation, NaN
paths), `test_common.py` (the shared decode/batching helper) and
`test_hpsv3_scorer.py` (`_resolve_checkpoint_path`: directory, file, repo id, bad
value). I ran them against *this exact tree* — dropped into `unirl-reward-service/tests/`
on this branch, `pytest tests/ -q` → **50 passed** — then removed them, since
`unirl-reward-service` carries no test tree upstream today and CI runs no pytest.
Say the word and I will push them onto this branch.

Real-hardware validation: an 8-GPU WAN2.1-1.3B T2V Flash-GRPO job driving this
service at `frame_stride: 1`, 8 replicas, FlashAttention-2 confirmed active on
the HPSv3 backbone, currently past rollout 8/1000 with no scorer errors. Frame
accounting was verified empirically — stride 1 decodes all 81 frames and
`top_ratio_mean` keeps 24.

## Compatibility / Risk

- **Additive for scorers.** No existing reward's score changes.
- **`hpsv3_scorer.py` changes one existing behaviour**: a `weights_path` that is
  neither a directory nor an existing file used to be handed to the inferencer as
  a file path (and fail there); it is now interpreted as a repo id and downloaded.
  Valid local paths behave exactly as before.
- **The new config downloads ~16.5 GB** from `MizzenAI/HPSv3` on first actor init.
  Nodes without egress should point `weights_path` at a local directory.
- **The FlashAttention-2 wheel URL is a hard platform pin**: `cp312` / `torch2.12` /
  `cu130`, Hopper `sm_90`. It cannot be a normal version spec — PyPI's `flash-attn`
  builds from source, and HPSv3's `create_model_and_processor` picks between FA2
  and SDPA purely on whether `import flash_attn` succeeds, so the presence of the
  package *is* the switch. FA3 does not substitute: its import name is
  `flash_attn_interface` and HPSv3 hardcodes the FA2 string. Reviewers on other
  platforms will need a different wheel. **This is the line I would most like a
  second opinion on** — it is correct for our nodes and fragile everywhere else.
- `trl` and `tensorboard` are new entries in `envs/hpsv3.txt` and are not
  optional: `hpsv3/__init__` → `inference.py:9` → `hpsv3/train.py:21` imports
  `trl` at module level, and `train.py:8` pulls in `qwen2vl_trainer`
  (`trl.RewardTrainer`, `torch.utils.tensorboard.SummaryWriter`). So they are
  import-time requirements even for inference-only scoring. Verified against the
  installed package: `trl 0.21.0 / tensorboard 2.21.0 / transformers 4.57.6`.
- No checkpoint, data-format or public API changes.

## Reviewer Notes

- **Draft on purpose.** The consumer recipe is not here: it targets
  `unirl.algorithms.flashgrpo.FlashGRPO`, which lands in #244, and the
  `recipe _target_ paths resolve` hook rejects it until that merges. This PR is
  self-contained and reviewable now (the scorer has zero `unirl` imports), but I
  would rather land it after or alongside #244 so the recipe can ship with it.
- `configs/videohpsv3_service.yaml` sets `num_replicas: 8` with a comment
  explaining that those GPUs are the same physical devices the trainer holds —
  deliberate over-subscription that fits HPSv3's ~17 GB of bf16 weights alongside
  a ~48 GB training footprint on a 97 GB H20. That is our deployment shape, not a
  universal default; happy to ship `1` if you prefer the checked-in config
  conservative.
- `score_timeout_s: 1800` looks enormous for one request, and it is not one
  request: `RewardService.score_and_attach` is DP_SCATTER-dispatched, every
  training rank POSTs its own shard, and `_await_ref` starts its clock at
  dispatch — so the deadline has to cover the whole rollout step.
- **Duplicate-work check**: no open PR touches `unirl-reward-service` video
  rewards. #269 (EditReward batching) is the only recent change in that package
  and touches different files; #142/#143/#146 added the image HPSv3 scorer and
  #97 added VideoAlign, both merged and both built on here rather than duplicated.
- **AI assistance**: authored with Claude Code. The full diff was reviewed and
  every command in the Test Plan was run locally before publication.
- Follow-up, not done here: `reward_service.config` does not `expandvars` on
  params, so a `weights_path` like `$HF_HOME/...` will not expand. The repo-id
  support removes most of the need, so I left it alone.

## Checklist

- [x] I reviewed the changed code and removed unrelated/generated artifacts.
- [x] I updated docs (`docs/ARCHITECTURE.md`) and configs. Tests exist and pass
      (50) but are deliberately not in this diff — see Test Plan.

### CODE REVIEW

**Comment by haonan3** [comment]
Before merging, please run an actual training job (for this reward, you need find proper training task & data) to validate the change. After integrating the VideoHPSv3 reward, confirm that the reward curve behaves normally. A screenshot of the curve or the corresponding training logs is sufficient.

Please convert the PR to draft for now and mark it as ready for review again once the validation results are available.
